server.js

```diff
@@ -1,23 +1,74 @@
 const express = require("express");
 const cors = require("cors");
+const { response, request } = require("express");
 
 const app = express();
-
+const bodyParser = require("body-parser")
 app.use(cors());
+app.use(express.json())
+app.use(bodyParser.json())
+//data
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
 };
 
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
+//---data
+
+
+
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
+app.get("/messages/search", (request, response) => {
+  const result = request.query.text;
+  let searchResult = messages;
+  if (result) {
+    searchResult.filter(item => item.text.includes(result))
+  }
+
+  response.send(searchResult)
+})
+app.get("/messages", (request, response) => {
+  response.send(messages)
+})
+
+app.get("/messages/:id", (request, response) => {
+  const idParam = Number(request.params.id);
+  const findObject = messages.filter(item => item.id === idParam)
+  response.send(findObject)
+})
```

Comment:
this works well when the given id exists, nice job. Have a think what you would send back if the id requested didn't exist (at the moment I'm getting a 200 and an empty array)